server/src/main/java/org/opensearch/common/xcontent/XContentHelper.java

```diff
@@ -87,7 +92,12 @@ public static XContentParser createParser(NamedXContentRegistry xContentRegistry
             if (compressedInput.markSupported() == false) {
                 compressedInput = new BufferedInputStream(compressedInput);
             }
-            return XContentFactory.xContent(xContentType).createParser(xContentRegistry, deprecationHandler, compressedInput);
+            try {
+                return XContentFactory.xContent(xContentType).createParser(xContentRegistry, deprecationHandler, compressedInput);
+            } catch (Exception e) {
+                compressedInput.close();
+                throw e;
+            }
```

Comment:
same as above 2 comments

Reply:
The InputStream looks like need to exist after the method returns, otherwise there will be test failure (see above), so I didn't modified this part.
I have triggered a new CodeGuru review to validate my changes above.